internal/tenant/index.go

```diff
@@ -0,0 +1,79 @@
+package tenant
+
+import (
+	"context"
+	"iter"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	proto "github.com/sourcegraph/zoekt/cmd/zoekt-sourcegraph-indexserver/protos/sourcegraph/zoekt/configuration/v1"
+	"github.com/sourcegraph/zoekt/internal/tenant/internal/tenanttype"
+)
+
+// ContextIndexDir returns a context and index dir for the given tenant ID.
+func ContextIndexDir(tenantID int, repoDir string) (context.Context, string) {
+	if !EnforceTenant() {
+		// Default to tenant 1 if enforcement is disabled.
+		return tenanttype.WithTenant(context.Background(), 1), repoDir
+	}
+	return tenanttype.WithTenant(context.Background(), tenantID), filepath.Join(repoDir, TenantsDir, strconv.Itoa(tenantID))
+}
```

Comment:
This is effectively a "give me tenant ID, receive context for it" method that we hard-don't use in sg/sg. Wondering if there's an easy enough way to work around this, or if the solution would be the push model we talked about. 

At the very least, we should add a `//🚨 SECURITY:` type comment here to say to not misuse this :)